code/modules/modular_computers/file_system/programs/alarm.dm

```diff
@@ -40,7 +40,7 @@
 	return data
 
 /datum/computer_file/program/alarm_monitor/proc/triggerAlarm(class, area/A, O, obj/source)
-	if(!is_station_level(source.z) && !is_mining_level(source.z))
+	if((!is_station_level(source.z) && !is_mining_level(source.z)) || !(A.type in GLOB.the_station_areas))
```

Comment:
Will only generate on the main z level
https://github.com/tgstation/tgstation/blob/8e692e6c02786b7b8e605b942d1568af5387abed/code/controllers/subsystem/mapping.dm#L209 This line only runs if its the station z. A proper solution is confirmation that all ruins use the /area/ruins pathing and istype from that.

Reply:
yes, that's exactly what I did at first. It's up to whatever @ShizCalev wants.

Reply:
That is a opinion I believe that Shiz doesnt know the station monitors are meant tomonitor the mining areas such as labor camp and general mining too.

Reply:
```
	if(is_station_level(source.z) && !(A.type in GLOB.the_station_areas))
		return
	else if(!is_mining_level(source.z) || istype(A, /area/ruin))
		return
```
my b., that'll fix it for station side ruins too.